### ISSUE TITLE
tf.nn.ctc_loss documentation misleads about GPU support, leading to 50x performance difference

### ISSUE THREAD

**Issue by psobot** [tensorflow/tensorflow#55290]
## URL(s) with the issue:

https://www.tensorflow.org/api_docs/python/tf/nn/ctc_loss

## Description of issue (what needs changing):

[The documentation for `tf.nn.ctc_loss`](https://github.com/tensorflow/tensorflow/blob/v2.8.0/tensorflow/python/ops/ctc_ops.py#L790) states:
> On TPU and GPU: Only dense padded labels are supported.
> On CPU: Caller may use SparseTensor or dense padded labels but calling with a SparseTensor will be significantly faster.

However, reading through [the code for the CTC loss implementation](https://github.com/tensorflow/tensorflow/blob/v2.8.0/tensorflow/python/ops/ctc_ops.py#L921-L957) shows that SparseTensors _are_ supported on GPU, and in practice, the SparseTensor code path uses CUDNN and is about 50x faster than the non-sparse code path.

I have tested this on GPU, but not on TPU, where I expect the documentation may be accurate (as the CUDNN code path doesn't apply to TPU). 

### Submit a pull request?

Would be happy to, after confirmation that this is indeed an issue.

cc @kaixih, @f90

**Comment by roniesha1994** [comment]
Sofware updating
